p9cpu/crates/libp9cpu/src/client/mod.rs

```diff
@@ -11,30 +11,49 @@ use std::fmt::Debug;
 #[async_trait]
 pub trait ClientInnerT {
     type Error: std::error::Error + Sync + Send + 'static;
+
+    /// The ID for one session, which corresponds to one command running on a
+    /// remote host.
     type SessionId: Clone + Debug + Sync + Send + 'static;
 
-    async fn dial(&mut self) -> Result<Self::SessionId, Self::Error>;
+    /// Talks to the remote machine and prepares for running a command.
+    async fn dial(&self) -> Result<Self::SessionId, Self::Error>;
 
-    async fn start(&mut self, sid: Self::SessionId, command: CommandReq)
-        -> Result<(), Self::Error>;
+    /// Starts the command on the remote machine. This method should return as
+    /// long as the command is spawned successfully. To obtained the command
+    /// return code, call [wait()](Self::wait).
+    async fn start(&self, sid: Self::SessionId, command: CommandReq) -> Result<(), Self::Error>;
 
     type EmptyFuture: Future<Output = Result<(), Self::Error>> + Send + 'static;
 
     type ByteVecStream: Stream<Item = Vec<u8>> + Unpin + Send + 'static;
-    async fn stdout(&mut self, sid: Self::SessionId) -> Result<Self::ByteVecStream, Self::Error>;
-    async fn stderr(&mut self, sid: Self::SessionId) -> Result<Self::ByteVecStream, Self::Error>;
 
+    /// Returns the stdout byte stream of a remote command. Callers `.await` the
+    /// stream to get the actual stdout bytes.
```

Comment:
do you have to wait until process exits (I hope not)

Reply:
No it's not required.